routesrv/redishandler.go

```diff
@@ -27,11 +27,17 @@ func (rh *RedisHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	w.Header().Add("Content-Type", "application/json")
-	w.Write(rh.AddrUpdater())
+	address, error := rh.AddrUpdater()
+	if error != nil {
+		log.Errorf("Could not update address for redis handler %v", error)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	w.Write(address)
 }
 
-func getRedisAddresses(namespace, name string, kdc *kubernetes.Client) func() []byte {
-	return func() []byte {
+func getRedisAddresses(namespace, name string, kdc *kubernetes.Client) func() ([]byte, error) {
+	return func() ([]byte, error) {
```

Comment:
error->err

Reply:
@szuecs the func return type should be of error type right ?

Reply:
@szuecs return type should be of error right ?

Reply:
yes you are right here. Just make sure that if you assign it to a var that you use `err` as var name.